version: "3.3"

services:
  admin:
    build:
      dockerfile: django_gift/Dockerfile
      context: .
    command: scripts/web/startup.sh -mm -m -l
#    volumes:
#       - ./django_gift:/code
    env_file:
      - django_gift/.env
    ports:
      - "8000:8000"
    depends_on:
      - main_db
      - redis
    networks:
      - gift_network

  bot:
    build:
      dockerfile: bot/Dockerfile
      context: .
    command: python main.py
    volumes:
      - ./bot:/code
    env_file:
      - bot/.env
    depends_on:
      - main_db
      - redis
      - admin
    networks:
      - gift_network

  main_db:
    image: postgres:13.2
    env_file:
      - .env.db
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "user", "-d", "postgres"]
      interval: 5s
      retries: 5
    restart: always
    ports:
      - "5432:5432"
    networks:
      - gift_network

  redis:
    image: redis:6.2.4
    networks:
      - gift_network


networks:
  gift_network:
    external:
      name: gift_network